src/Microsoft.ML.FastTree/BinFile/BinFinder.cs

```diff
@@ -43,46 +44,51 @@ public BinFinder()
         /// The scheme is destructive, because it modifies the arrays within <paramref name="values"/>.
         /// </summary>
         /// <param name="values">The values we are binning</param>
+        /// <param name="valueBuffer">A buffer space to work over the values, so the original
+        /// values aren't modified.</param>
         /// <param name="distinctValues">This working array will be filled with a sorted list of the
         /// distinct values detected within <paramref name="values"/></param>
         /// <param name="counts">This working array will be filled with a sorted list of the distinct
         /// values detected within <paramref name="values"/></param>
         /// <returns>The logical length of both <paramref name="distinctValues"/> and
         /// <paramref name="counts"/></returns>
-        private int FindDistinctCounts(in VBuffer<Double> values, double[] distinctValues, int[] counts)
+        private int FindDistinctCounts(in VBuffer<Double> values, double[] valueBuffer, double[] distinctValues, int[] counts)
         {
-            if (values.Count == 0)
+            var valueValues = values.GetValues();
```

Comment:
>valueValues [](start = 16, length = 11)

`valueValues` goes against feng shui.
Maybe `explicitValues` or something?

Reply:
Changed to `explicitValues` and `explicitValuesCount`

---
In reply to: [233672186](https://github.com/dotnet/machinelearning/pull/1580#discussion_r233672186) [](ancestors = 233672186)